UndertaleModTool/Controls/UndertaleRoomRenderer.xaml.cs

```diff
@@ -64,7 +64,9 @@ private void RoomRenderer_DataContextChanged(object sender, DependencyPropertyCh
             {
                 if (layer.LayerType == LayerType.Assets)
                 {
-                    var particleSystems = layer.AssetsData.ParticleSystems.Select(x => x.ParticleSystem);
+                    var particleSystems = Enumerable.Empty<UndertaleParticleSystem>();
+                    if (layer.AssetsData.ParticleSystems is not null)
+                        particleSystems = layer.AssetsData.ParticleSystems.Select(x => x.ParticleSystem);
                     ParticleSystemRectConverter.Initialize(particleSystems);
```

Comment:
I think it should be like this:
```suggestion
                    if (layer.AssetsData.ParticleSystems is not null)
                    {
                        var particleSystems = layer.AssetsData.ParticleSystems.Select(x => x.ParticleSystem);
                        ParticleSystemRectConverter.Initialize(particleSystems);
                    }
```

Reply:
So this part is executed only if `layer.AssetsData.ParticleSystems` is not null

Reply:
No, there is no point to do that - if the room doesn't contain any particle system instances, then then converter wouldn't be used for that room.

Reply:
But I think we still should change this line just in case:
https://github.com/krzys-h/UndertaleModTool/blob/64b52a65ee4b0e8eecfa59ebf0a440eac52f55e3/UndertaleModTool/Editors/UndertaleRoomEditor.xaml.cs#L2694
(make it `if (!partSystemsDict?.TryGetValue(...`)

Reply:
changed these, not sure if it was what you wanted.

Reply:
Almost, see the latest suggestion